models/benchmarks/himeno_model.py

```diff
@@ -92,13 +92,20 @@ def build_benchmark(self, compilers_dict, complete_compile_flags,
     def run_benchmark(self, binary_name, extra_runflags):
         """Runs the benchmarks using the base + extra flags"""
         binary_name = 'bmt'
-        binary_path = os.path.join(self.build_root, binary_name)
-        run_cmd = [[]]
-        if extra_runflags is None or extra_runflags == '':
-            run_cmd.append([binary_path])
-        else:
-            run_cmd.append([binary_path, extra_runflags])
-        return run_cmd
+        binary_path = os.path.join(
+            self.benchmark_rootpath, self.name, binary_name)
+        run_cmds = []
+        run_args = self.parse_run_args(extra_runflags)
+        if (run_args.iterations).isdigit():
+            for i in range(0, int(run_args.iterations)):
+                run_cmd = []
+                metadata = 'i=' + str(i) + 's=' + \
```

Comment:
I'm not sure why you're passing the arguments serialised like this and then de-serialise as command line arguments above.

Reply:
It's up to the benchmark to interpret each option, except iteration.
So each benchmark has it's own parser for its own special needs.
Makes more sense to do it like that imo. If you have a better idea please suggest.